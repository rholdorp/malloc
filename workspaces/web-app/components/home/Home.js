import React, { Component } from 'react'
import { FadingValueBox } from '../animations'
import { AssignmentDialog } from '../assignments'
import { TeamMemberDialog } from '../team-members'

const getCurrentlySignedUser = () => {
  const user = firebase.auth().currentUser
  if (user) {
    return user.uid
  } else {
    return null
  }
}

const createAssignment = async assignment => {
  try {
    const uid = getCurrentlySignedUser()
    if (uid) {
      const { assignmentName, projectNumber } = assignment
      const db = firebase.firestore()
      await db.collection('assignments').add({
        assignmentName,
        projectNumber
      })
    } else {
      throw new Error('Having trouble accesing Firebase. Please try again...')
    }
  } catch (e) {
    console.error(e)
  }
}

const createTeamMember = async teamMember => {
  try {
    const uid = getCurrentlySignedUser()
    if (uid) {
      const { name, availability } = teamMember
      const db = firebase.firestore()
      await db.collection('teamMembers').add({
        name,
        availability
      })
    } else {
      throw new Error('Having trouble accesing Firebase. Please try again...')
    }
  } catch (e) {
    console.error(e)
  }
}

// const read = async tag => {
//   const uid = getCurrentlySignedUser()
//   if (uid) {
//     const db = firebase.firestore()
//     const doc = await db.collection('users').doc(uid).get()
//     const senderTag = doc.get(tag)

//   } else {
//     throw new Error('Having trouble accesing Firebase. Please try again...')
//   }
// }
class Home extends Component {
  state = {
    assignmentDialogOpen: false,
    teamMemberDialogOpen: false
  }

  addAssignment = async assignment => {
    this.setState({ assignmentDialogOpen: false })
    await createAssignment(assignment)
  }

  addTeamMember = async teamMember => {
    this.setState({ teamMemberDialogOpen: false })
    await createTeamMember(teamMember)
  }

  openAssignmentDialog = () => {
    this.setState({ assignmentDialogOpen: true })
  }

  onCancelAddingAssignment = () => {
    this.setState({ assignmentDialogOpen: false })
  }

  openTeamMemberDialog = () => {
    this.setState({ teamMemberDialogOpen: true })
  }

  onCancelAddingTeamMember = () => {
    this.setState({ teamMemberDialogOpen: false })
  }

  render () {
    return (
      <FadingValueBox>
        <AssignmentDialog open={this.state.assignmentDialogOpen}
          buttonText='Add assignment...'
          buttonStyling={{ basic: true, color: 'black' }}
          onOpen={this.openAssignmentDialog}
          onDone={this.addAssignment}
          onCancel={this.onCancelAddingAssignment} />
        <TeamMemberDialog open={this.state.teamMemberDialogOpen}
          buttonText='Add team member...'
          buttonStyling={{ basic: true, color: 'black' }}
          onOpen={this.openTeamMemberDialog}
          onDone={this.addTeamMember}
          onCancel={this.onCancelAddingTeamMember} />
      </FadingValueBox>
    )
  }
}

export { Home }
