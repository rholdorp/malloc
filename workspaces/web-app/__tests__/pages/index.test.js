import React from 'react'
import { render } from 'react-testing-library'
import Home from '../../pages/index'

describe('index', () => {
  beforeEach(() => {
    jest.useFakeTimers()
  })

  it('has proper styling', () => {
    const { container } = render(<Home />)
    // for the fading animation to finish
    jest.runAllTimers()
    expect(container).toMatchSnapshot()
  })

  it('displays something', () => {
    const { getByText } = render(<Home />)
    // for the fading animation to finish
    jest.runAllTimers()
    expect(getByText(/add assignment/i)).toBeInTheDocument()
  })
})
