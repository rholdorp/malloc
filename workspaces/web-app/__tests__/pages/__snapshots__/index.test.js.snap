// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index has proper styling 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100vh;
  width: 100vw;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  opacity: 1;
  -webkit-transition: opacity 1s ease-in-out 0s;
  transition: opacity 1s ease-in-out 0s;
}

<div>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      css="[object Object]"
    >
      <div
        class="emotion-0 emotion-1"
        opacity="1"
      >
        <button
          class="ui black basic button"
        >
          Add assignment...
        </button>
        <button
          class="ui black basic button"
        >
          Add team member...
        </button>
      </div>
    </div>
  </div>
</div>
`;
